controllers/managedocs_controller.go

```diff
@@ -319,9 +320,9 @@ func (r *ManagedOCSReconciler) initReconciler(req ctrl.Request) {
 	r.deadMansSnitchSecret.Name = r.DeadMansSnitchSecretName
 	r.deadMansSnitchSecret.Namespace = r.namespace
 
-	r.alertmanagerConfigSecret = &corev1.Secret{}
-	r.alertmanagerConfigSecret.Name = alertmanagerConfigSecretName
-	r.alertmanagerConfigSecret.Namespace = r.namespace
+	r.alertmanagerConfig = &promv1a1.AlertmanagerConfig{}
+	r.alertmanagerConfig.Name = alertmanagerConfigSecretName
```

Comment:
Why do use the secret name? The secret name ends with the word "secret" which does not fit here. 
I will suggest using a name that follows our convention: `managed-ocs-alertmanager-config` in a constant named `alertmanagerConfigName`